src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -83,13 +91,18 @@ public DataWriteService(ReactiveCqlTemplate cqlTemplate,
     final String seriesSetHash = seriesSetService.hash(metric.getMetric(), metric.getTags());
 
     return storeRawData(tenant, metric, seriesSetHash)
-        .name("ingest")
-        .metrics()
-        .and(metadataService.storeMetadata(tenant, seriesSetHash, metric.getMetric(), metric.getTags()))
-        .and(downsampleTrackingService.track(tenant, seriesSetHash, metric.getTimestamp()))
-        .and(storeMetricGroup(tenant, metric))
-        .and(storeDeviceData(tenant, metric))
-        .then(Mono.just(metric));
+            .name("ingest")
+            .metrics()
+            .and(metadataService.storeMetadata(tenant, seriesSetHash, metric.getMetric(), metric.getTags()))
+            .and(downsampleTrackingService.track(tenant, seriesSetHash, metric.getTimestamp()))
+            .and(storeMetricGroup(tenant, metric))
+            .and(storeDeviceData(tenant, metric))
+            .then(Mono.just(metric))
+            .doOnNext(_metric -> recordIngestionLatency(_metric));
+  }
+
+  private void recordIngestionLatency(Metric metric) {
+    latencyTimer.record(Duration.between(metric.getTimestamp(), Instant.now()).getSeconds(), TimeUnit.SECONDS);
```

Comment:
So this will record the duration between the time at which the metrics was created to the time it was ingested?

Reply:
Correct. Thats my understanding as well. Each metric has timestamp of metric creation. Using that to calculate the duration till ingestion.